Bump Istio to 1.27.0 (#1578)
kyma-project/istio: docs/release-notes/1.21.0.md

## Istio Updated to Version 1.27.0

We've updated the Istio version to 1.27.0 ([#1578](https://github.com/kyma-project/istio/pull/1578)).
Read the [Istio 1.27.0 release announcement](https://istio.io/latest/news/releases/1.27.x/announcing-1.27/) as well as [Upgrade Notes](https://istio.io/latest/news/releases/1.27.x/announcing-1.27/upgrade-notes/) to learn about the new features, bug fixes, and breaking changes in this release.

> [!Warning]

> > Istio since version 1.27.0 injects the Istio proxy as an initContainer by default. In this release, the operator disables this behavior to maintain compatibility. Please note that this disablement is temporary and will be removed in future releases, after which proxies will be injected as initContainers by default.

Reviewer: ```suggestion
> Since version 1.27.0, Istio injects the Istio proxy as an initContainer by default. In this release, the Istio module's operator disables this behavior to maintain compatibility. Disabling this behavior is a temporary measure and will be removed in one of the future releases. After this, sidecar proxies will be injected as initContainers by default.
```
